### PR TITLE
Fix events flyout bugs and Styling

### PR DESCRIPTION
**Pull request by ps48**
### Description
fix events flyout bugs and styling issues.

### Issues Resolved
1. Styling changes w.r.t z-index
2. Added Back button to event details
3. Disabled surroundings events w/o time field
4. Updated PPL time format with more precision 
5. Updated jest test for doc viewer
6. Added Trace Hyperlink 

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by mengweieric — DISMISSED** [review]

File: dashboards-observability/public/components/explorer/docTable/docViewRow.tsx [hunk 1]
```diff
@@ -36,6 +35,12 @@ export const DocViewRow = (props: IDocViewRowProps) => {
   const [detailsOpen, setDetailsOpen] = useState<boolean>(false);
   const [surroundingEventsOpen, setSurroundingEventsOpen] = useState<boolean>(false);
   const [openTraces, setOpenTraces] = useState<boolean>(false);
+  const [flyoutToggleSize, setFlyoutToggleSize] = useState(true);
+  const [logTraceId, setLogTraceId] = useState('');
+
+  const isValidTraceId = (traceId: string) => {
```
Comment (joshuali925) [inline dashboards-observability/public/components/explorer/docTable/docViewRow.tsx:41]: why redefine it here? also why 32? i don't think it's fixed

Comment (eugenesk24) [inline dashboards-observability/public/components/explorer/docTable/docViewRow.tsx:41]: Isn't this function already in the utils.tsx??

**Review by eugenesk24 — DISMISSED** [review]

File: dashboards-observability/public/components/explorer/docTable/docViewRow.tsx (hunk 1 above)
Comment (ps48) [reply dashboards-observability/public/components/explorer/docTable/docViewRow.tsx:41]: Thanks for the catch. The 32 is declared from https://www.w3.org/TR/trace-context/#trace-id, got the link from open-telemetry docs.

Comment (ps48) [reply dashboards-observability/public/components/explorer/docTable/docViewRow.tsx:41]: Thanks will remove the duplicate

Comment (ps48) [reply dashboards-observability/public/components/explorer/docTable/docViewRow.tsx]: @joshuali925 If it seems unnecessary, I can remove it.

**Review by joshuali925 — APPROVED** [review]

**Review by mengweieric — APPROVED** [review]

File: dashboards-observability/public/components/explorer/docTable/docViewRow.tsx (hunk 1 above)
Comment (joshuali925) [reply dashboards-observability/public/components/explorer/docTable/docViewRow.tsx]: should be ok. I was thinking about shorter trace IDs but they will be padded with leading zeros in data-prepper

**Pull request merged**